internal/host/validator.go

```diff
@@ -795,3 +795,42 @@ func (v *validator) printSufficientPacketLossRequirementForRole(c *validationCon
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) isDomainNameResolvedCorrectly(c *validationContext) ValidationStatus {
+	var response models.DomainResolutionResponse
+	if err := json.Unmarshal([]byte(c.host.APIDomainNameResolution), &response); err != nil {
+		v.log.WithError(err).Warn("Parse domain resolution")
+		return ValidationError
+	}
+	if len(response.Resolutions) == 0 {
+		return ValidationFailure
+	}
+	for _, domain := range response.Resolutions {
+		if domain.DomainName == c.cluster.APIVipDNSName {
+			for _, ipv4 := range domain.IPV4Addresses {
+				if ipv4 == strfmt.IPv4(c.cluster.APIVip) {
+					return ValidationSuccess
+				}
+			}
+			for _, ipv6 := range domain.IPV6Addresses {
+				if ipv6 == strfmt.IPv6(c.cluster.APIVip) {
+					return ValidationSuccess
+				}
+			}
+		}
+	}
+	return ValidationFailure
+}
+
+func (v *validator) printIsDomainNameResolvedCorrectly(c *validationContext, status ValidationStatus) string {
+	switch status {
+	case ValidationSuccess:
+		return "API name resolution was successful"
+	case ValidationFailure:
+		return fmt.Sprintf("Failed to resolve the %s domain correctly on the host. Expected domain to resolve to IP address %s. The installation will not be able to complete as long as this host's DNS server is unable to resolve this domain correctly. Please create the necessary DNS entries to continue", *c.cluster.APIVipDNSName, c.cluster.APIVip)
```

Comment:
- @avishayt please review this / come up with a better message - I mostly wrote this message as a draft, didn't mean for it to get copied as-is.

- It would be nice if this message also contained the actual IP the domain was actually resolved to, or alternatively, if the domain was not resolved at all, the message should specify that as well.